### DISCUSSION TITLE
Access to petite-vue  from outside

### DISCUSSION THREAD

**Discussion by Alexufo** [vuejs/petite-vue#45] [General]
Is there any method that can put data to petite outside components? Like in vue `var vm = new Vue({...}); console.log(vm.$data.name);`?

**Comment by nooooooom** [comment]
As far as I know, all data except `initialData` will be parsed after `mounted`, so you can write custom `directives` instructions to collect the context of each node. For the simplest example

```js
<div id="app" v-scope="{name: 'Tom'}" v-collect></div>

<script type="module">
  import { createApp } from '../src'

  const ctxMap = new WeakMap()

  const collectCtx = dirCtx => {
    ctxMap.set(dirCtx.el, dirCtx.ctx)
  }

  createApp()
    .directive('collect', collectCtx)
    .mount('#app')
</script>

```

**Reply by Alexufo** [reply]
thx - I finally found example on main page about global state :-) I use it :-)
https://github.com/vuejs/petite-vue#global-state-management
